app/models/gene.rb

```diff
@@ -68,6 +68,11 @@ def species
   # CLASS INSTANCE METHODS
   ##
 
+  # find if a study has a given gene quickly for search gating
+  def self.study_has?(study_id, expr_matrix_ids, gene_name)
```

Comment:
Nice, I like how you pulled this into a separate method.  How about renaming to `study_has_gene?` for clarity?

Also, I could see using named parameters here so that mixing up argument order is less likely